lib/Controller/FolderController.php

```diff
@@ -145,10 +146,15 @@ private function getRootFolderStorageId(): ?int {
 	/**
 	 * @RequireGroupFolderAdmin
 	 * @NoAdminRequired
+	 * @throws OCSNotFoundException
 	 */
 	public function addFolder(string $mountpoint): DataResponse {
 		$id = $this->manager->createFolder(trim($mountpoint));
-		return new DataResponse(['id' => $id]);
+		$folder = $this->manager->getFolder($id, $this->rootFolder->getMountPoint()->getNumericStorageId());
+		if ($folder === false) {
+			throw new OCSNotFoundException();
+		}
+		return new DataResponse($folder);
```

Comment:
This changes the response type, is `$folder` containing `'id'`?

Reply:
Yes